designcompose/src/main/java/com/android/designcompose/DesignFrame.kt

```diff
@@ -124,10 +132,7 @@ internal fun DesignFrame(
         return
     }
 
-    // Check to see if this node is part of a component set with variants and if any @DesignVariant
-    // annotations set variant properties that match. If so, variantNodeName will be set to the
-    // name of the node with all the variants set to the @DesignVariant parameters
-    val variantNodeName = customizations.getMatchingVariant(componentInfo)
+    // If we a custom variant, compose it instead and then return.
     if (variantNodeName != null) {
         // Get the generated CustomVariantComponent() function and call it with variantNodeName
         val customComposable = customizations.getCustomComposable()
```

Comment:
Is it possible for `variantNodeName` to be non-null, but `customComposable` is null? Then I think we'd render the node without its background (but maybe this is an error and we should do something else?)

Reply:
It was possible before, which was the first part of the fix in the `mergeFrom()` function. From the commit:
"There was a problem where the customComposable function did not propagate all customizations down to its children properly: specifically, it did not propagate the customComposable() function itself. This has been fixed in the CustomizationContext.mergeFrom() function."

Now, it *shouldn't* be possible anymore because `customizations.setCustomComposable()` is called in generated code, and anytime we pass this down to a child it gets copied to the child customizations.

Reply:
I don't know if there's a cleaner way to express the logic, because it takes a bit of reasoning to get there, but this makes sense to me.

Reply:
Ok, this logic may be moving to Rust soon, so I'm going to leave it for now.